catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -193,13 +203,27 @@ public Metacard getMetacard(String id)
         }
 
         Filter queryFilter = filterBuilder.anyOf(filters);
-        QueryResponse response = catalogFramework.query(new QueryRequestImpl(new QueryImpl(
-                queryFilter,
-                1,
-                -1,
-                SortBy.NATURAL_ORDER,
-                false,
-                TimeUnit.SECONDS.toMillis(10)), false));
+        QueryRequest queryRequest;
+
+        if (CollectionUtils.isEmpty(writableSources)) {
+            queryRequest = new QueryRequestImpl(new QueryImpl(
+                    queryFilter,
+                    1,
+                    querySize,
+                    SortBy.NATURAL_ORDER,
+                    false,
+                    TimeUnit.SECONDS.toMillis(10)), false);
+        } else {
+            queryRequest = new QueryRequestImpl(new QueryImpl(
+                    queryFilter,
+                    1, querySize,
+                    SortBy.NATURAL_ORDER,
+                    false,
+                    TimeUnit.SECONDS.toMillis(10)), writableSources);
```

Comment:
✏️ / ❗️ Timeout for a federated query we probably should move to 30 seconds (or maybe 5 minutes or whatever the default is for DDF, though I'm fine pushing that work into a different ticket as well so we can get this merged.)